osf/metadata/serializers.py

```diff
@@ -0,0 +1,136 @@
+import json
+from datacite import schema40
+
+from osf.metadata import utils
+from website.settings import DOMAIN
+
+serializer_registry = {}
+
+def register(schema_id):
+    """Register classes into serializer_registry"""
+    def decorator(cls):
+        serializer_registry[schema_id] = cls
+        return cls
+    return decorator
+
+
+class MetadataRecordSerializer(object):
+
+    def serialize_json(self, metadata_record):
+        raise NotImplementedError
+
+    def serialize_xml(self, metadata_record):
+        raise NotImplementedError
+
+    @classmethod
+    def serialize(cls, metadata_record, format='json'):
+        if format == 'json':
+            return cls.serialize_json(metadata_record)
+        if format == 'xml':
+            return cls.serialize_xml(metadata_record)
+        raise ValueError('Format "{}" is not supported.'.format(format))
+
+
+@register(schema_id='datacite')
+class DataciteMetadataRecordSerializer(MetadataRecordSerializer):
+
+    osf_schema = 'osf_datacite.json'
+
+    @classmethod
+    def serialize_json(cls, record):
```

Comment:
I know I shouldn't criticize because I wrote this method (I think), but this is long and should be broke up if just for clarity sake. We might want to standardize this somewhat because XML etc will be much harder to serialize.